### PR TITLE
Task 2

### PR DESCRIPTION
**Pull request by SamoylenkVi**

### CODE REVIEW

File: project/src/components/DetailsNavigation/helper.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export const ucFirst = (str:string) => str[0].toUpperCase() + str.slice(1);
```
Comment (Pavel-Liteiniy) [inline project/src/components/DetailsNavigation/helper.ts]: Let's rename to more understandable

File: project/src/components/DetailsNavigation/index.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import { ucFirst } from './helper';
+import { FilmDetailsContent } from '../FilmDetailsContent';
+import {Film} from '../../types/film';
+
+type DetailsNavigationProps = {
+  film: Film;
+}
+
+export const DetailsNavigation = ({ film }:DetailsNavigationProps) => {
+  const [currentButton, setCurrentButton] = useState<string>(NAVIGATION_BUTTONS.DETAILS);
```
Comment (Pavel-Liteiniy) [inline project/src/components/DetailsNavigation/index.tsx]:
```suggestion
  const [currentButton, setCurrentButton] = useState(NAVIGATION_BUTTONS.DETAILS);
```

Does ts by itself understand that it's string?

File: project/src/components/FilmList/index.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  const [, setActiveCardId] = useState('');
+
+  const handleMouseEnter = (id:string) => {
+    setActiveCardId(id);
+  };
+
+  return (
+    <>
+      <div className="catalog__films-list">
+        {Array.from(
```
Comment (Pavel-Liteiniy) [inline project/src/components/FilmList/index.tsx]: Can we use just `films.map` here?

File: project/src/components/FilmReview/constants.ts [hunk 4]
```diff
@@ -0,0 +1,2 @@
+export const DATA_FORMAT_TEXT = 'MMMM D, YYYY';
```
Comment (Pavel-Liteiniy) [inline project/src/components/FilmReview/constants.ts]: let's move date formats to global constants

File: project/src/components/FilmReview/helper.ts [hunk 5]
```diff
@@ -0,0 +1,4 @@
+import { default as dayjs } from 'dayjs';
+
+export const formatData = (data:Date, format:string) => dayjs(data).format(format);
```
Comment (Pavel-Liteiniy) [inline project/src/components/FilmReview/helper.ts]: lets move to global helpers

File: project/src/components/Footer/index.tsx [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+    <div className="logo">
+      <a className="logo__link logo__link--light">
+        <span className="logo__letter logo__letter--1">W</span>
+        <span className="logo__letter logo__letter--2">T</span>
+        <span className="logo__letter logo__letter--3">W</span>
+      </a>
+    </div>
+
+    <div className="copyright">
+      <p>© 2019 What to watch Ltd.</p>
```
Comment (Pavel-Liteiniy) [inline project/src/components/Footer/index.tsx]: let's replace hardcoded year to autogenerated with dayjs

File: project/src/components/PrivateRoute/index.tsx [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import { Navigate } from 'react-router';
+import { AUTHORIZATION_STATUS, APP_ROUTE } from '../../constants';
+
+type PrivateRouteProps = {
+  authorizationStatus: 'AUTH' | 'NO_AUTH' | 'UNKNOWN';
```
Comment (Pavel-Liteiniy) [inline project/src/components/PrivateRoute/index.tsx]: let's linked `authorizationStatus` prop and `AUTHORIZATION_STATUS` with using this approach

```
const STATUS = {
    AUTH: 'auth',
    DENY: 'deny',
    OK: 'ok',
} as const;

type key = keyof typeof STATUS
type Status = typeof STATUS[key]
```

File: project/src/pages/FilmDetailsPage/index.tsx [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+                  </svg>
+                  <span>Play</span>
+                </button>
+                <button className="btn btn--list film-card__button" type="button">
+                  <svg viewBox="0 0 19 20" width="19" height="20">
+                    <use xlinkHref="#add"></use>
+                  </svg>
+                  <span>My list</span>
+                </button>
+                {(id) ? <Link to={`/films/${id}/review`} className="btn film-card__button">Add review</Link> : ''}
```
Comment (Pavel-Liteiniy) [inline project/src/pages/FilmDetailsPage/index.tsx]:
```suggestion
                {(id !== undefined) && (
                  <Link to={`/films/${id}/review`} className="btn film-card__button">Add review</Link> 
                )}
```

File: project/src/pages/MainPage/index.tsx [hunk 9]
```diff
@@ -0,0 +1,80 @@
+import { FilmCardMain } from '../../components/FilmCardMain';
+import { FilmList } from '../../components/FilmList';
+import { Footer } from '../../components/Footer';
+
+type Film = {
```
Comment (Pavel-Liteiniy) [inline project/src/pages/MainPage/index.tsx]: Is it same type as Film in `project/src/types/film.ts`? Should we delete it?

File: project/src/types/comment.ts [hunk 10]
```diff
@@ -0,0 +1,10 @@
+export type Comment = {
+  id: number;
+  user: {
```
Comment (Pavel-Liteiniy) [inline project/src/types/comment.ts]: Let's add separated type for user

File: project/src/utils/FilmCard.ts
Comment (Pavel-Liteiniy) [inline project/src/utils/FilmCard.ts:1]: let's rename this file to `createAltText`

**Review by Pavel-Liteiniy — CHANGES_REQUESTED** [review]

File: project/src/components/DetailsNavigation/helper.ts (hunk 1 above)
Comment (SamoylenkVi) [reply project/src/components/DetailsNavigation/helper.ts]: done

File: project/src/components/DetailsNavigation/index.tsx (hunk 2 above)
Comment (SamoylenkVi) [reply project/src/components/DetailsNavigation/index.tsx]: done

File: project/src/components/FilmList/index.tsx (hunk 3 above)
Comment (SamoylenkVi) [reply project/src/components/FilmList/index.tsx]: done

File: project/src/components/FilmReview/constants.ts (hunk 4 above)
Comment (SamoylenkVi) [reply project/src/components/FilmReview/constants.ts]: done

File: project/src/components/FilmReview/helper.ts (hunk 5 above)
Comment (SamoylenkVi) [reply project/src/components/FilmReview/helper.ts]: done

File: project/src/components/Footer/index.tsx (hunk 6 above)
Comment (SamoylenkVi) [reply project/src/components/Footer/index.tsx]: done

File: project/src/components/PrivateRoute/index.tsx (hunk 7 above)
Comment (SamoylenkVi) [reply project/src/components/PrivateRoute/index.tsx]: done

File: project/src/pages/FilmDetailsPage/index.tsx (hunk 8 above)
Comment (SamoylenkVi) [reply project/src/pages/FilmDetailsPage/index.tsx]: done

File: project/src/pages/MainPage/index.tsx (hunk 9 above)
Comment (SamoylenkVi) [reply project/src/pages/MainPage/index.tsx]: done

File: project/src/types/comment.ts (hunk 10 above)
Comment (SamoylenkVi) [reply project/src/types/comment.ts]: done

File: project/src/utils/FilmCard.ts
Comment (SamoylenkVi) [reply project/src/utils/FilmCard.ts:1]: done

**Comment by Pavel-Liteiniy** [comment]
Approved! Good job 🔥

**Pull request merged**